viadot/sources/s3.py

```diff
@@ -130,46 +140,88 @@ def from_df(
+            max_rows_by_file (int): Max number of rows in each file (only works for read_parquet). Default to None.
         """
 
-        if path.endswith(".csv"):
+        if extension == ".csv":
             wr.s3.to_csv(
                 boto3_session=self.session,
                 df=df,
                 path=path,
-                dataset=True,
                 **kwargs,
             )
-        elif path.endswith(".parquet"):
+        else:
             wr.s3.to_parquet(
                 boto3_session=self.session,
                 df=df,
                 path=path,
-                dataset=True,
+                max_rows_by_file=max_rows_by_file,
                 **kwargs,
             )
-        else:
-            raise ValueError("Only CSV and parquet formats are supported.")
 
     def to_df(
         self,
-        path: str,
+        paths: list[str],
+        chunked: Union[int, bool] = False,
         **kwargs,
     ):
         """
         Reads a csv or parquet file to a pd.DataFrame.
 
         Args:
-            path (str): Path to a S3 folder.
+            paths (list[str]): A list of paths to S3 files.
+            chunked (Union[int, bool], optional): If True data will be split in a Iterable of DataFrames (Memory friendly).
+                If Integer data will be intereated by number of rows equal to the received Integer.
+
+        Example 1:
+        ```python
+        from viadot.sources import S3
+        s3 = S3()
+        s3.to_df(paths=['s3://{bucket}/pathfirstfile.parquet', 's3://{bucket}/pathsecondfile.parquet'])
+        ```
+
+        Example 2:
+        ```python
+        from viadot.sources import S3
+        s3 = S3()
+        dfs = s3.to_df(paths=['s3://{bucket}/pathfirstfile.parquet', 's3://{bucket}/pathsecondfile.parquet'], chunked = True/Integer)
+        for df in dfs:
+            print(df)
+        ```
         """
-        if path.endswith(".csv"):
+
+        if paths[0].endswith(".csv"):
             df = wr.s3.read_csv(
-                boto3_session=self.session, path=path, dataset=True, **kwargs
+                boto3_session=self.session, path=paths, chunked=chunked, **kwargs
             )
-        elif path.endswith(".parquet"):
+        elif paths[0].endswith(".parquet"):
             df = wr.s3.read_parquet(
-                boto3_session=self.session, path=path, dataset=True, **kwargs
+                boto3_session=self.session, path=paths, chunked=chunked, **kwargs
             )
         else:
             raise ValueError("Only CSV and parquet formats are supported.")
         return df
+
+    def upload(self, from_path: str, to_path: str):
+        """
+        Upload file(s) to S3.
+
+        Args:
+            from_path (str): Path to local file(s) to be uploaded.
+            to_path (str): Path to the destination file/folder.
+        """
+
+        wr.s3.upload(boto3_session=self.session, local_file=from_path, path=to_path)
+
+    def download(self, to_path: str, from_path: str):
```

Comment:
nit: `from_path` should be first IMO

Reply:
Updated in:

- [📝 Changed Parameter Order](https://github.com/dyvenia/viadot/pull/603/commits/93a9fdc8761f71116585b9a6a3f2abfa5db1ceb6)